src/main/java/com/quizapp/enterprise/services/GameService.java

```diff
@@ -75,9 +80,16 @@ private boolean userNameExists(String userName, String gameCode){
 
     }
 
-    public GuessResult ProcessPlayerGuess(String userGuess, String gameCode, Long questionId, String playerName) {
+    public GuessResult ProcessPlayerGuess(String userGuess,
+                                          String gameCode,
+                                          Long questionId,
+                                          String playerName) throws Exception {
 
         Question question = questionRepository.getById(questionId);
+        if(question == null){
+            throw new Exception("question not found");
+        }
+
```

Comment:
included a check for a possible null prior to execution on possible null object so code may handle the case as desired.
This change was recommended by the "8. Proper handling of Null Pointer Exceptions"  section of  the java coding best practices summary referenced below

source:https://xperti.io/blogs/java-coding-best-practices/